Mathlib/Analysis/Convex/Cone/Proper.lean

```diff
@@ -30,77 +32,107 @@ The next steps are:
 
 -/
 
-open ContinuousLinearMap Filter Set
+open ContinuousLinearMap Filter Function Set
 
-/-- A proper cone is a pointed cone `K` that is closed. Proper cones have the nice property that
+variable {R E F G : Type*} [Semiring R] [PartialOrder R] [IsOrderedRing R]
+variable [AddCommMonoid E] [TopologicalSpace E] [Module R E]
+variable [AddCommMonoid F] [TopologicalSpace F] [Module R F]
+variable [AddCommMonoid G] [TopologicalSpace G] [Module R G]
+
+local notation "R≥0" => {r : R // 0 ≤ r}
+
+variable (R E) in
+/-- A proper cone is a pointed cone `C` that is closed. Proper cones have the nice property that
 they are equal to their double dual, see `ProperCone.dual_dual`.
 This makes them useful for defining cone programs and proving duality theorems. -/
-structure ProperCone (𝕜 : Type*) (E : Type*)
-    [Semiring 𝕜] [PartialOrder 𝕜] [IsOrderedRing 𝕜] [AddCommMonoid E]
-    [TopologicalSpace E] [Module 𝕜 E] extends Submodule {c : 𝕜 // 0 ≤ c} E where
-  isClosed' : IsClosed (carrier : Set E)
+abbrev ProperCone := ClosedSubmodule R≥0 E
 
 namespace ProperCone
 section Module
+variable {C C₁ C₂ : ProperCone R E} {r : R} {x : E}
 
-variable {𝕜 : Type*} [Semiring 𝕜] [PartialOrder 𝕜] [IsOrderedRing 𝕜]
-variable {E : Type*} [AddCommMonoid E] [TopologicalSpace E] [Module 𝕜 E]
+/-- Alias of `ClosedSubmodule.toSubmodule` for convenience and discoverability. -/
+@[coe] abbrev toPointedCone (C : ProperCone R E) : PointedCone R E := C.toSubmodule
 
-/-- A `PointedCone` is defined as an alias of submodule. We replicate the abbreviation here and
-define `toPointedCone` as an alias of `toSubmodule`. -/
-abbrev toPointedCone (C : ProperCone 𝕜 E) := C.toSubmodule
+instance : Coe (ProperCone R E) (PointedCone R E) := ⟨toPointedCone⟩
 
-attribute [coe] toPointedCone
+lemma toPointedCone_injective : Injective ((↑) : ProperCone R E → PointedCone R E) :=
+  ClosedSubmodule.toSubmodule_injective
 
-instance : Coe (ProperCone 𝕜 E) (PointedCone 𝕜 E) :=
-  ⟨toPointedCone⟩
+-- TODO: add `ConvexConeClass` that extends `SetLike` and replace the below instance
+instance : SetLike (ProperCone R E) E where
+  coe C := C.carrier
+  coe_injective' _ _ h := ProperCone.toPointedCone_injective <| SetLike.coe_injective h
 
-theorem toPointedCone_injective : Function.Injective ((↑) : ProperCone 𝕜 E → PointedCone 𝕜 E) :=
-  fun S T h => by cases S; cases T; congr
+@[ext] lemma ext (h : ∀ x, x ∈ C₁ ↔ x ∈ C₂) : C₁ = C₂ := SetLike.ext h
 
--- TODO: add `ConvexConeClass` that extends `SetLike` and replace the below instance
-instance : SetLike (ProperCone 𝕜 E) E where
-  coe K := K.carrier
-  coe_injective' _ _ h := ProperCone.toPointedCone_injective (SetLike.coe_injective h)
+@[simp] lemma mem_toPointedCone : x ∈ C.toPointedCone ↔ x ∈ C := .rfl
```

Comment:
`mem_coe` makes more sense as the name for this, since in the infoview it'll get printed as a coercion (this is also consistent with the old way)

Reply:
The reason I rename it is that there are a priori several coercions from `ProperCone`: to `PointedCone`, to `ConvexCone`, to `Set`